kiosk/sources/extensions/owned_kiosk.move

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Description:
+///
+module kiosk::owned_kiosk {
+    use std::option::{Self, Option};
+    use sui::transfer;
+    use sui::kiosk::{Self, Kiosk, KioskOwnerCap};
+    use sui::object::{Self, ID, UID};
+    use sui::tx_context::{sender, TxContext};
+    use sui::dynamic_field as df;
+
+    const EIncorrectCapObject: u64 = 0;
+    const EIncorrectOwnedObject: u64 = 1;
+
+    /// A key-only wrapper for the KioskOwnerCap. Makes sure that the Kiosk can
+    /// not be traded altogether with its contents.
+    struct OwnedKiosk has key {
+        id: UID,
+        cap: Option<KioskOwnerCap>
+    }
+
+    /// The hot potato making sure the KioskOwnerCap is returned after borrowing.
+    struct Borrow { cap_id: ID, owned_id: ID }
+
+    /// The dynamic field to mark the Kiosk as owned (to allow guaranteed owner
+    /// checks through the Kiosk).
+    struct OwnerMarker has copy, store, drop {}
+
+    /// Wrap the KioskOwnerCap making the Kiosk "owned" and non-transferable.
+    public fun new(kiosk: &mut Kiosk, cap: KioskOwnerCap, ctx: &mut TxContext) {
+        let owner = sender(ctx);
+
+        // set the owner property of the Kiosk
+        kiosk::set_owner(kiosk, &cap, ctx);
```

Comment:
Random question: what is the purpose of the `owner` field in the kiosk? It seems like we gate all owner functionality with `KioskOwnerCap`

Reply:
It was added there following the request from OB folks with potential use case for logical ownership.

Reply:
What is logical ownership?

Reply:
Well, let's just say this field has no purpose but cosmetics + up to users to decide.